src/App.js

```diff
@@ -1,16 +1,85 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import ColorChoice from './components/ColorChoice';
 
 const App = () => {
+  const[entriesData, setEntriesData] = useState(chatMessages);
+
+  const likeEntry = (id) => {
+    setEntriesData((prev)=>{
+      return prev.map((entry)=>{
+        if(id === entry.id){
+          return{
+            ...entry,
+            liked: !entry.liked
+          };}
```

Comment:
👍 Nice use of spreading and object shorthand to make a new copy of the message being updated, while also toggling its liked status.